product_configurator/models/product_config.py

```diff
@@ -582,10 +578,36 @@ def update_session_configuration_value(self, vals, product_tmpl_id=None):
                 custom_val_dict.update({attr_id: val})
                 # Ensure there is no standard value stored if we have switched
                 # from selected value to custom value.
-                attr_val_dict.update({attr_id: False})
+                attr_val_dict.update({attr_id: custom_val.id})
 
         self.update_config(attr_val_dict, custom_val_dict)
 
+    def _update_field_values(self, vals, field_name, attr_line):
+        """New method for update field values for a given attribute."""
+        final_val = None
+        if not vals[field_name]:
+            return final_val
+        # Retrieve existing values for the attribute
+        value_ids = self.value_ids.filtered(
+            lambda value, attr_line=attr_line: value.attribute_id.id
+            == attr_line.attribute_id.id
+        )
+        # Initialize `final_val` with IDs
+        final_val = value_ids.ids or []
+
+        # Process each `field_vals` operation for the current attribute
+        for field_vals in vals.get(field_name, []):
+            if field_vals and field_vals[0] == Command.SET:
```

Comment:
Those are not triggered from the onchange web client, so even if we add code for those cases, we will not have a proper way to reproduce and test them.

Reply:
I see. It's a bit fragile, though.
If, for some reason, Odoo refactors the onchange and starts outputting other commands, this would fail.

I wonder if we can use the field's `convert_to_cache` method instead. ([many](https://github.com/odoo/odoo/blob/f2e295cd4ebc48e371bd226f5714069b3e127a1c/addons/sale/models/payment_transaction.py#L215) [examples](https://github.com/odoo/odoo/blob/cd4b18a46f8748157e767f265e777b7f4dc53422/addons/account/models/account_move.py#L2236) [in odoo code](https://github.com/odoo/odoo/blob/a431b4ab58f8c9a3fc7df59a1be194ba8d47fe24/addons/loyalty/models/loyalty_program.py#L438))
That method offers the advantage of not having to worry about the internals of onchange nor commands

Reply:
I see. Thanks for taking the time to explain all this.